src/ISComm.c

```diff
@@ -883,20 +899,24 @@ int is_comm_free(is_comm_instance_t* c)
             // we will be hung unless we flush the ring buffer, we have to drop bytes in this case and the caller
             // will need to resend the data
             parseErrorResetState(c, EPARSE_RXBUFFER_FLUSHED);
-            buf->head =
-            buf->tail =
-            buf->scan = buf->start;
+            return is_comm_reset_buffer(c, buf->start, buf->size);
         }
         else
         {	// Shift current data to start of buffer
             memmove(buf->start, buf->head, buf->tail - buf->head);
             buf->head = buf->start;
             buf->tail -= shift;
             buf->scan -= shift;
-        }
 
-        // re-calculate free byte count
-        bytesFree = (int)(buf->end - buf->tail);
+            // re-calculate free byte count
+            bytesFree = (int)(buf->end - buf->tail);
+        }
+    }
+    else if (c->processPkt == NULL && buf->scan == buf->tail)
```

Comment:
Can we change this to `else if (c->processPkt == NULL && buf->scan >= buf->tail)`